src/helpers.js

```diff
@@ -873,3 +873,36 @@ export function getIfaceSourceName(iface) {
 export function canDeleteDiskFile(disk) {
     return disk.type === "volume" || (disk.type === "file" && disk.source.file);
 }
+
+export function getStoragePoolPath(storagePools, poolName, connectionName) {
+    const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === connectionName);
+
+    return pool?.target?.path;
+}
+
+export function vmSupportsExternalSnapshots(config, vm, storagePools) {
+    const disks = Object.values(vm.disks);
+
+    // If at leat one disk has internal snapshot preference specified, use internal snapshot for all disk,
+    // as mixing internal and external is not allowed
+    if (disks.some(disk => disk.snapshot === "internal"))
+        return false;
+
+    // Currently external snapshots for only for disks where we can specify a local path of a newly created disk snapshot file
+    // This rules out all disks which are not file-based, or pool-based where pool is of type "dir"
+    if (!disks.every(disk => {
+        const supportedPools = storagePools.filter(pool => pool.type === "dir" && pool.connectionName === vm.connectionName);
+
+        return disk.type === "file" ||
+            (disk.type === "volume" && supportedPools.some(pool => pool.name === disk.source.pool));
+    }))
+        return false;
+
+    // External snapshot should onyl be used if the VM's os types, architecture allows it
+    // and if snapshot features are present among guest capabilities:
+    // https://libvirt.org/formatcaps.html#guest-capabilities
+    return config.capabilities?.guests.some(guest => guest.osType === vm.osType &&
+                                                     guest.arch === vm.arch &&
+                                                     guest.features.diskSnapshot &&
+                                                     guest.features.externalSnapshot);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20231030-133824-a492f2d5-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)